src/core/ipv4/ip4_napt.c

```diff
@@ -86,47 +92,67 @@ struct portmap_table {
 };
 
 static u16_t napt_list = NO_IDX, napt_list_last = NO_IDX, napt_free = 0;
+static u8_t ip_portmap_max = 0;
+
+static struct ip_napt_entry *ip_napt_table = NULL;
+static struct ip_portmap_entry *ip_portmap_table = NULL;
 
-static struct napt_table *ip_napt_table = NULL;
-static struct portmap_table *ip_portmap_table = NULL;
+static struct ip_napt_stats napt_stats;
 
-static int nr_active_napt_tcp = 0, nr_active_napt_udp = 0, nr_active_napt_icmp = 0;
-static uint16_t ip_napt_max = 0;
-static uint8_t ip_portmap_max = 0;
+static void ip_napt_gc(uint32_t now, bool force);
+static void ip_napt_tmr(void *arg);
+
+#define WRAPPED_AROUND(a, b) ((((a) ^ (b)) & (1UL << 31)) != 0)
 
 #if NAPT_DEBUG
-/* Print NAPT table using LWIP_DEBUGF
-*/
+/* Print NAPT table using LWIP_DEBUGF */
+#define DPRINTF(m) LWIP_DEBUGF(NAPT_DEBUG, m)
+/* #define DPRINTF(m) printf m */
 static void
 napt_debug_print(void)
 {
   int i, next;
-  LWIP_DEBUGF(NAPT_DEBUG, ("NAPT table:\n"));
-  LWIP_DEBUGF(NAPT_DEBUG, (" src                     dest                    sport   dport   mport   \n"));
-  LWIP_DEBUGF(NAPT_DEBUG, ("+-----------------------+-----------------------+-------+-------+-------+\n"));
-
+  u32_t now = sys_now();
+  uint16_t nr_total = napt_stats.nr_active_tcp + napt_stats.nr_active_udp + napt_stats.nr_active_icmp;
```

Comment:
Nit + mixed declaration
```suggestion
  u8_t nr_total = ...
```